content/raw/chat/mutual_contacts.md

```diff
@@ -0,0 +1,78 @@
+---
+title: "Mutual Contact Requests spec"
+description: ""
+date: 2020-10-06T08:48:57+00:00
+lastmod: 2020-10-06T08:48:57+00:00
+draft: false
+images: []
+menu:
+  raw:
+    parent: "chat"
+weight: 100
+toc: true
+---
+
+# Mutual Contact Requests
+
+## Motivation
+
+To prevent unsolicited messages & invitations and some forms of spam, Users must explicity add each other as contacts.
+
+## Designs
+
+* [designs](https://www.figma.com/file/Mr3rqxxgKJ2zMQ06UAKiWL/%F0%9F%92%AC-Chat%E2%8E%9CDesktop?node-id=5224%3A0)
+* [Sending contact request](https://www.figma.com/file/IPpvkpDWabBKJTeo6bFop0/Kuba%E2%8E%9CDesktop?node-id=725%3A329626)
+* [Receiving contact request](https://www.figma.com/file/IPpvkpDWabBKJTeo6bFop0/Kuba%E2%8E%9CDesktop?node-id=1637%3A369041)
+
+## Use Cases
+
+### Sending a Contact Request
+
+1. User selects a profile and asks application sends contact request
+2. User fills reason for contact request e.g "Say who you are / why you want to become a contact.."
+3. User sends contact request
+4. System notifies user `Contact Request Send`
+
+### Receiving Contact Request
+
+1. System notifies user of a contact request (note: see activity center spec)
+2. User checks pending contact requests
+3. system displays the username of the requester, time of request, and the motivation text for the request
+4. User accepts contact request
+4.1 alternative: User rejects contact request
+5. system makes contact request as accepted or rejected
```

Comment:
shouldn't it remove the contact request from the list once it's accepted or rejected, how/when does that happen?

Reply:
In the Activity Centre, an accepted or rejected contact request should be treated the same way as an @mention message in the Activity Centre that has been read.  By this I mean that if the user has selected 'hide read messages' then accepted and rejected contact requests should be hidden, and if the user has selected 'show read messages' then then accepted and rejected contact requests should be displayed